apps/tangle-dapp/data/NominatorStats/usePaymentDestinationSubscription.ts

```diff
@@ -0,0 +1,67 @@
+'use client';
+
+import { WebbError, WebbErrorCodes } from '@webb-tools/dapp-types/WebbError';
+import { useEffect, useState } from 'react';
+import { type Subscription } from 'rxjs';
+
+import { getPolkadotApiRx } from '../../constants';
+import useFormatReturnType from '../../hooks/useFormatReturnType';
+
+export default function usePaymentDestinationSubscription(
+  address: string,
+  defaultValue: { value1: number | string | null } = { value1: null }
+) {
+  const [value1, setValue1] = useState(defaultValue.value1);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    let isMounted = true;
+    let sub: Subscription | null = null;
+
+    const subscribeData = async () => {
+      try {
+        const api = await getPolkadotApiRx();
+        if (!api) {
+          throw WebbError.from(WebbErrorCodes.ApiNotReady);
```

Comment:
We're throwing this but do we catch it when this hook is used? Why do we need this try/catch here then? You're kinda using try/catches everywhere but that's making the code even more unclean.